SceneRefAttributeValidator.cs

```diff
@@ -222,48 +225,67 @@ private static object UpdateRef(SceneRefAttribute attr, Component c, FieldInfo f
                             ? (existingValue as ISerializableRef[])?.Select(existingRef => GetComponentIfWrongType(existingRef.SerializedObject, elementType)).ToArray()
                             : GetComponentIfWrongType(existingValue, elementType);
                     break;
+
                 case RefLoc.Self:
                     value = isArray
-                        ? c.GetComponents(elementType)
-                        : c.GetComponent(elementType);
+                        ? (object)c.GetComponents(elementType)
+                        : (object)c.GetComponent(elementType);
                     break;
+
                 case RefLoc.Parent:
+
+#if UNITY_2020
+                    value = isArray
+                        ? (object)c.GetComponentsInParent(elementType, includeInactive)
+                        : (object)c.GetComponentInParent(elementType);
+#endif
+#if UNITY_2021_OR_NEWER
                     value = isArray
-                        ? c.GetComponentsInParent(elementType, includeInactive)
-                        : c.GetComponentInParent(elementType, includeInactive);
+                        ? (object)c.GetComponentsInParent(elementType, includeInactive)
+                        : (object)c.GetComponentInParent(elementType, includeInactive);
+#endif
+
                     break;
+
                 case RefLoc.Child:
                     value = isArray
-                        ? c.GetComponentsInChildren(elementType, includeInactive)
-                        : c.GetComponentInChildren(elementType, includeInactive);
+                        ? (object)c.GetComponentsInChildren(elementType, includeInactive)
+                        : (object)c.GetComponentInChildren(elementType, includeInactive);
                     break;
+
                 case RefLoc.Scene:
+#if UNITY_2020_3_26
```

Comment:
This version is very specific, any reason not to use `UNITY_2020` here?

Reply:
No not really, this is the specific version I was working in. This should be for all 2020 versions